packages/utils/src/ts-eslint/Rule.ts

```diff
@@ -8,7 +8,14 @@ import type { SourceCode } from './SourceCode';
 
 export type RuleRecommendation = 'recommended' | 'strict' | 'stylistic';
 
-export interface RuleMetaDataDocs {
+export interface RuleRecommendationAcrossConfigs<
+  Options extends readonly unknown[],
+> {
+  recommended: true;
+  strict: Partial<Options>;
+}
```

Comment:
🤔 Part of me doesn't like that we're leaking our internal documentation implementation details into our public types.

I wonder if there's a way for us to privatise these types to avoid leaking them to users?

Reply:
That would be nice. I don't know though 😞 and it's already there.

Reply:
Filed #8695.